contracts/strategy/BnbxYieldConverterStrategy.sol

```diff
@@ -0,0 +1,245 @@
+    /// @return value - returns the amount of BNB withdrawn from stader (currently 0 as no amount is withdrawn in this step)
+    function _withdraw(address recipient, uint256 amount)
+        internal
+        returns (uint256 value)
+    {
+        require(amount > 0, "invalid amount");
+
+        uint256 bnbxAmount = _stakeManager.convertBnbToBnbX(amount);
+        _bnbDepositBalance -= amount;
+        _bnbxHoldingBalance -= bnbxAmount;
+
+        _bnbxToUnstake += bnbxAmount;
+        _withdrawRequests[_nextWithdrawIdx++] = UserWithdrawRequest({
+            recipient: recipient,
+            amount: amount,
+            bnbxAmount: bnbxAmount,
+            triggerTime: block.timestamp
+        });
+
+        return 0;
+    }
+
+    // actual withdraw request to stader
+    function batchWithdraw() external {
+        require(_bnbxToUnstake > 0, "No BNBx to unstake");
+
+        uint256 bnbxToUnstake = _bnbxToUnstake; // To prevent reentrancy
+        _bnbxToUnstake = 0;
+        _stakeManager.requestWithdraw(bnbxToUnstake);
+    }
+
+    /// @dev claims the next available withdraw batch from stader
+    /// @dev transfer funds(BNB) from stakeManager to strategy
+    /// @return foundClaimableReq : true if claimed any batch, false if no batch is available to claim
+    function claimNextBatch() external returns (bool foundClaimableReq) {
+        uint256 claimableIdx;
+        bool foundClaimableReq = false;
+
+        IStakeManager.WithdrawalRequest[] memory requests = _stakeManager
+            .getUserWithdrawalRequests(address(this));
+
+        for (uint256 idx = 0; idx < requests.length; idx++) {
+            (bool isClaimable, uint256 amount) = _stakeManager
+                .getUserRequestStatus(address(this), idx);
+            if (isClaimable) {
+                foundClaimableReq = true;
+                claimableIdx = idx;
+                _bnbToDistribute += amount;
+                break;
+            }
+        }
+
+        if (!foundClaimableReq) return foundClaimableReq;
+
+        _stakeManager.claimWithdraw(claimableIdx);
+        return foundClaimableReq;
+    }
+
+    /// @dev distribute claimed funds to users in FIFO order of withdraw requests
+    /// @param maxNumRequests : parameter to control max number of requests to refund
+    /// @return reqCount : actual number requests refunded
+    function distributeFund(uint256 maxNumRequests)
+        external
+        returns (uint256 reqCount)
+    {
+        reqCount = 0;
+
+        while (
+            _firstWithdrawIdx < _nextWithdrawIdx &&
+            _withdrawRequests[_firstWithdrawIdx].amount <= _bnbToDistribute
+        ) {
+            if (reqCount == maxNumRequests) break;
+            address recipient = _withdrawRequests[_firstWithdrawIdx].recipient;
+            uint256 amount = _withdrawRequests[_firstWithdrawIdx].amount;
+
+            delete _withdrawRequests[_firstWithdrawIdx];
+            _firstWithdrawIdx++;
+
+            _bnbToDistribute -= amount;
```

Comment:
Would this automatically error out if bnbToDistribute < amount?

Reply:
already checked above 
`amount <= _bnbToDistribute`